src/dbnode/server/options.go

```diff
@@ -27,7 +27,8 @@ import (
 
 // StorageOptions are options to apply to the database storage options.
 type StorageOptions struct {
-	OnColdFlush            storage.OnColdFlush
-	ForceColdWritesEnabled bool
-	TChanNodeServerFn      node.NewTChanNodeServerFn
+	OnColdFlush             storage.OnColdFlush
+	ForceColdWritesEnabled  bool
+	TChanNodeServerFn       node.NewTChanNodeServerFn
+	AfterNamespaceCreatedFn storage.AfterNamespaceCreatedFn
```

Comment:
I'm more in favor of using interfaces than callbacks.

i.e.

```go
type NamespaceHooks interface {
  OnCreatedNamespace(....)
}
```

That way future methods/events can be added to the interface without extra plumbing of an extra "FooFn" being plumed from start to finish.

Thoughts?

Reply:
Agree 100%, updating.